cli/src/commands/parse.rs

```diff
@@ -251,8 +249,36 @@ fn filter_packages(mut packages: Vec<Package>) -> Vec<PackageDescriptor> {
         .collect()
 }
 
+/// Find a file by walking from a directory towards the root.
+///
+/// `MAX_DEPTH` is the maximum number of directory traversals before the search
+/// will be abandoned. A `MAX_DEPTH` of `0` will only search the `origin`
+/// directory.
+fn find_direntry_upwards<const MAX_DEPTH: usize, P>(
+    mut origin: &Path,
+    mut predicate: P,
+) -> Option<PathBuf>
+where
+    P: FnMut(&Path) -> bool,
+{
+    for _ in 0..=MAX_DEPTH {
+        for entry in fs::read_dir(origin).ok()?.flatten() {
+            let path = entry.path();
+            if predicate(&path) {
```

Comment:
Is it a problem that this would also find a directory named `Cargo.lock`?

The previous code also didn't distinguish files or directories (or symlinks). So we can ignore it for this PR. And I don't **think** it will cause any real world issues, so maybe we don't need to address it at all.

Reply:
Honestly that should probably be checked in the `is_path_lockfile` functions if anything.

Feel free to open an issue about this if we ever have absolutely **nothing** else to do, but I think this scenario is **very** unlikely to ever cause problems.